core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -73,6 +73,28 @@ coreos:
 
         [Install]
         RequiredBy=format-etcd2-volume.service
+{{if and (.AmazonSsmAgent.Enabled) (ne .AmazonSsmAgent.DownloadUrl "")}}
+    - name: amazon-ssm-agent.service
+      command: start
+      enable: true
+      content: |
+        [Unit]
+        Description=amazon-ssm-agent
+        Requires=network-online.target
+        After=network-online.target
+
+        [Service]
+        Type=simple
+        WorkingDirectory=/
+        ExecStartPre=/opt/ssm/bin/install-ssm-agent.sh
+        ExecStart=/opt/ssm/bin/amazon-ssm-agent
+        KillMode=process
```

Comment:
I am not familiar with Amazon SSM agent, but is there a reason for this KillMode? According to docs it kills top level process only, leaving anything it spawns to stay alive.